chapters/traversal.md

```diff
@@ -1,83 +1,89 @@
+
+[//]: # (The original link below does not make sense any more)
+[//]: # (https://hanxiao.io/2020/11/22/Primitive-Data-Types-in-Neural-Search-System/)
+
+Following the definition of [primitive data types in Jina](primitive_data_type.rst), a chunk is a property of a Document:
 
-In Jina [primitive data types](https://hanxiao.io/2020/11/22/Primitive-Data-Types-in-Neural-Search-System/), `chunk` is defined as a `property` of a Document:
 
 ```python
 from jina import Document
 
 with Document() as root:
     root.text = 'What is love? Oh baby do not hurt me.'
-# Initialised a document as root with 0 chunks.
-print(len(root.chunks))
->>> 0
-# Initialise two documents and add as chunks to root.
+
+# Initialised a Document as root with 0 chunks.
+print(len(root.chunks))                # outputs 0
+
+# Initialise two Documents and add them as chunks to root.
 with Document() as chunk1:
     chunk1.text = 'What is love?'
     root.chunks.add(chunk1)
 with Document() as chunk2:
     chunk1.text = 'Oh baby do not hurt me.'
     root.chunks.add(chunk2)
-# Now the document has 2 chunks
-print(len(root.chunks))
->>> 2
+
+# Now the Document has 2 chunks
+print(len(root.chunks))                # outputs 2
 ```
 
-What happened by adding `chunk` to `root`?
+### Internals: Adding a chunk to a root node
+
+Now let us have a look at what happens when you added a chunk to a root node. The code example below shows the granularity level of a specific node using the attribute `granularity` -- as a root node (level 0), and as a chunk (level 1).
 
 ```python
-print(root.granularity)
->>> 0
-print(root.chunks[0].granularity)
->>> 1
-root.id == root.chunks[0].parent_id
->>> True
+print(root.granularity)                # outputs 0
+print(root.chunks[0].granularity)      # outputs 1
 ```
-
-This can be seen in the image below:
+This is illustrated in the image below:
 
 ![granularity](./images/granularity.png)
 
-The code sample and graph above demonstrates the basic idea of a `chunk` in a Document.
-In the beginning, we initialized a Document with `granularity=0` (by default).
-We then initialized two chunks and add them to the `root` document.
-Two things happened when adding the `chunk` to `root`:
+Furthermore, with the help of the attribute `parent_id` the relation between the root node and the chunk can be validated.
+
+```python
+print(root.id == root.chunks[0].parent_id)  # outputs True
+```
+
+The code sample and graph above demonstrates the basic idea of a `chunk` in a Document. At the beginning, by default a Document is initialised with `granularity = 0`. Next, two chunks have been initialised with `granularity = 0`, and have been added to the root node. Adding a chunk to a node, two things happen:
```

Comment:
nit: change `Adding` to `On addition of`

Reply:
I corrected that, too